src/utils/rangeMatchModifiers.ts

```diff
@@ -0,0 +1,123 @@
+import { defaultDateLib, type DateLib } from "../classes/DateLib.js";
+import type { DateRange, Matcher } from "../types/index.js";
+
+import { dateMatchModifiers } from "./dateMatchModifiers.js";
+import { rangeIncludesDate } from "./rangeIncludesDate.js";
+import {
+  isDateAfterType,
+  isDateBeforeType,
+  isDateInterval,
+  isDateRange,
+  isDatesArray,
+  isDayOfWeekType
+} from "./typeguards.js";
+
+/**
+ * Returns whether a date range matches against at least one of the given
+ * {@link Matcher}.
+ *
+ * ```tsx
+ * const range: DateRange = {
+ *  from: new Date(2021, 12, 21),
+ *  to: new Date(2021, 12, 30)
+ * }
+ * const matcher1: Date = new Date(2021, 12, 21)
+ * const matcher2: DateRange = {
+ *  from: new Date(2022, 5, 1),
+ *  to: new Date(2022, 5, 23)
+ * }
+ * const dateMatchModifiers(date, [matcher1, matcher2]); // true, since matcher1 is in the date.
+ * ```
+ *
+ * @group Utilities
+ */
+export function rangeMatchModifiers(
```

Comment:
I think the function should be renamed;

```suggestion
export function rangeContainsModifiers(
```

Reply:
FYI this function follows the same pattern as `dateMatchModifiers`, it felt more appropriate to follow the same naming because it's the same idea but applied for date ranges.
I'll proceed with applying the suggestion in any case, if you change your mind lemme know.